Result/ResultType.swift

```diff
@@ -51,6 +51,18 @@ public extension ResultType {
 			ifFailure: Result<U, Error>.Failure)
 	}
 	
+	/// Returns a new Result by mapping `Success`es’ values using `transform`, or wrapping errors thrown as `Failure`s’.
+	public func tryMap<U>(@noescape transform: Value throws -> U) -> Result<U, Error> {
+        return flatMap { value in
+            do {
+                return .Success(try transform(value))
+            }
+            catch {
+                return .Failure(error as! Error)
```

Comment:
@robrix Can you provide a hint how to accomplish such a constrained extension? 
Also, how is this worse than the existing initializer? 
public init(@autoclosure _ f: () throws -> T) {
		do {
			self = .Success(try f())
		} catch {
			self = .Failure(error as! Error)
		}
	}